File: java-pi-pantry/src/test/java/com/langrsoft/util/JSonUtilToJson.java
package com.langrsoft.util;

import com.langrsoft.util.JsonParseException;
import com.langrsoft.util.JsonUtil;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;

public class JSonUtilToJson {
    @Test
    public void convertsObjectToString() {
        assertThat(JsonUtil.toJson(new Object() { public int x = 42; }),
                is(equalTo("{\"x\":42}")));
    }

    @Test(expected=JsonParseException.class)
    public void throwsRuntimeExceptionOnJsonProcessingException() {
        JsonUtil.toJson(new Object());
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/UtilTests/JSonUtilToJson.cs
using NUnit.Framework;
using Pipantry.Domain;
using Pipantry.Util;
using System;

namespace Test.Pipantry.Util
{
    public class JSonUtilToJson
    {
        class SomeClass
        {
            public SomeClass() { X = 42; }
            public int X { get; }
        }

        [Test]
        public void ConvertsObjectToString()
        {
            Assert.That(JsonUtil.ToJson(new SomeClass()),
                Is.EqualTo("{\"X\":42}"));
        }

        class CannotSerialize
        {
            public int X { get { throw new ApplicationException(); } }
        }

        [Test]
        public void ThrowsRuntimeExceptionOnJsonProcessingException()
        {
            Assert.Throws<JsonParseException>(() => JsonUtil.ToJson(new CannotSerialize()));
        }
    }
}
